pkg/utils/annotations/annotations.go

```diff
@@ -42,3 +43,27 @@ func GetTargetNameFromAnnotation(
 
 	return "", fmt.Errorf("neither %q annotation nor '_raw' suffix found", targetAnnotation)
 }
+
+// SetFieldVisibility sets the visibility of a field using metadatav1.ColumnsHiddenAnnotation
+// For a field with subfields, it will only set the visibility of the root field
+func SetFieldVisibility(
+	defaultValue bool,
+	in datasource.FieldAccessor,
+) {
+	annotations := in.Annotations()
+
+	in.SetHidden(true, true)
+
+	outName, ok := annotations[metadatav1.ColumnsHiddenAnnotation]
+	if !ok {
+		in.SetHidden(defaultValue, false)
+	} else {
+		hide := false
+
+		if outName == "true" {
+			hide = true
+		}
+
+		in.SetHidden(hide, false)
```

Comment:
very minor nit while looking around. (I won't fully review it)

```suggestion
		in.SetHidden(outName == "true", false)
```